packages/cadl-typescript/src/modelUtils.ts

```diff
@@ -905,3 +907,41 @@ export function getFormattedPropertyDoc(
   }
   return propertyDoc ?? enhancedDocFromType;
 }
+
+export function getBodyType(program: Program, route: HttpOperation) {
+  let bodyModel = route.parameters.bodyType;
+  if (bodyModel && bodyModel.kind === "Model" && route.operation) {
+    const resourceType = getResourceOperation(
```

Comment:
I notice that it's hard to get the body type, not sure you know some context and then we could add some comments for why we do this way.

Reply:
In the case where we are using resource operation from cadl-azure-core, there's a resource type, if they don't use any custom wrapper we can use resourceType to replace the body type, but in the customization case, we need to keep the body type not replace it.